main.go

```diff
@@ -72,6 +72,25 @@ func main() {
 	elasticsearchProxy := func(w http.ResponseWriter, r *http.Request) {
 		proxyURL := elasticBackendURL + "/" + r.URL.Path[15:]
 		fmt.Println("proxy:", proxyURL)
+			logstash1, err := appEnv.Services.WithTag("logstash")
+			if err == nil {
+				fmt.Println("getting username and password")
+				username, ok := logstash1[0].Credentials["username"]
+				if ok {
+					username = logstash1[0].Credentials["username"].(string)
+				}
```

Comment:
For this, you risk a panic if the `logstash1[0].Credentials["username"]` value is not a string. To avoid that, you can use:

```
var username string
var password string
var ok bool
if username, ok = logstash1[0].Credentials["username"].(string); !ok {
   fmt.Println("Unsupported non-string value found for ElasticSearch username")
}
if password, ok = logstash1[0].Credentials["password"].(string); !ok {
   fmt.Println("Unsupported non-string value found for ElasticSearch password")
}
```